sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -960,17 +961,52 @@ case class GpuRegExpReplace(
   override def second: Expression = searchExpr
   override def third: Expression = replaceExpr
 
-  def this(srcExpr: Expression, searchExpr: Expression, cudfRegexPattern: String,
-      cudfReplacementString: String) = {
-    this(srcExpr, searchExpr, GpuLiteral("", StringType), cudfRegexPattern, cudfReplacementString)
+  def this(srcExpr: Expression, searchExpr: Expression, javaRegexpPattern: String,
+    cudfRegexPattern: String, cudfReplacementString: String) = {
+    
+    this(srcExpr, searchExpr, GpuLiteral("", StringType), javaRegexpPattern, 
+      cudfRegexPattern, cudfReplacementString)
   }
 
   override def doColumnar(
       strExpr: GpuColumnVector,
       searchExpr: GpuScalar,
       replaceExpr: GpuScalar): ColumnVector = {
-    withResource(Scalar.fromString(cudfReplacementString)) { rep =>
-      strExpr.getBase.replaceRegex(cudfRegexPattern, rep)
+
+    val isEmptyRepetition: Boolean = {
```

Comment:
Should this be a recursive check using the recently added `contains` method? I'm thinking cases like `(((A*)))`